src/nvim/os/input.c

```diff
@@ -323,13 +325,25 @@ static unsigned int handle_mouse_event(char **ptr, uint8_t *buf,
   return bufsize;
 }
 
+bool input_blocking(void)
+{
+  return blocking;
+}
+
 static bool input_poll(int ms)
 {
   if (do_profiling == PROF_YES && ms) {
     prof_inchar_enter();
   }
 
+  if ((ms == - 1 || ms > 0)
+      && !(events_enabled || input_ready() || input_eof)
```

Comment:
Because it has been quite some time since I've maintained this code, I'm not sure what are the implications of adding this `multiqueue_process_events()` call, but consider that `input_poll` is called no matter what state nvim is in, so it may not be safe to process events.

In any case, this condition is the same as `(ms == - 1 || ms > 0) && !events_enabled && !input_ready() && !input_eof`, but `input_ready()` cannot be true for this function to be called(See `input_poll` below), so it could be simplified to `(ms == - 1 || ms > 0) && !events_enabled && !input_eof`